implementation/src/main/java/io/smallrye/metrics/app/WeightedSnapshot.java

```diff
@@ -67,14 +67,18 @@ public WeightedSnapshot(Collection<WeightedSample> values) {
         this.normWeights = new double[copy.length];
         this.quantiles = new double[copy.length];
 
-        double sumWeight = 0;
+        double sumWeight = 0.0D;
         for (WeightedSample sample : copy) {
             sumWeight += sample.weight;
         }
 
         for (int i = 0; i < copy.length; i++) {
             this.values[i] = copy[i].value;
-            this.normWeights[i] = copy[i].weight / sumWeight;
+            if (sumWeight == 0.0D || Double.compare(sumWeight, 0.0D) == 0) {
```

Comment:
Is it right that both comparisons are checking `sumWeight` to be `0.0D`?

Reply:
i think need use Double.compare, but sonar think ==. Results of these comparisons maybe difference.

Reply:
I'm not absolutely sure how these comparisons work, but it looks like defensive programming, so I'd say it's ok :)